tools/sctool/src/main/java/com/horizen/ScBootstrappingToolCommandProcessor.java

```diff
@@ -393,27 +402,40 @@ private void processGenerateCertWithKeyRotationProofInfo(JsonNode json) throws E
                 return;
             }
 
-            mastersPublicKeys.add(pkNode.asText());
+            String masterKey = pkNode.asText();
+
+            int duplicateIndex = mastersPublicKeys.indexOf(masterKey);
+            if (duplicateIndex != -1) {
+                printGenerateCertWithKeyRotationProofInfoUsageMsg(String.format("duplicated keys in master keys are found. MasterKey with index %d equals to masterKey with index %d", duplicateIndex + 1, index));
+                return;
+            }
+
+            mastersPublicKeys.add(masterKey);
+            index++;
         }
 
-        assert mastersPublicKeys.size() == signersPublicKeys.size() : "mastersPublicKeys and signersPublicKeys must have the same size";
+        if (mastersPublicKeys.size() != signersPublicKeys.size()) {
+            printGenerateCertWithKeyRotationProofInfoUsageMsg(String.format("an amount of the signer keys and the master keys must be equal."));
```

Comment:
This error could be clearer:

"the number of signer keys must be equal to the number of master keys"